crates/air-lib/interpreter-cid/src/verify.rs

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use crate::{value_json_hash, CidRef, CID, JSON_CODEC};
+
+use serde::Serialize;
+use thiserror::Error as ThisError;
+
+use std::convert::TryInto;
+use std::rc::Rc;
+
+#[derive(ThisError, Debug)]
+pub enum CidVerificationError {
+    #[error("Value mismatch in the {type_name:?} store for CID {cid_repr:?}")]
+    MismatchError {
```

Comment:
I think we can avoid suffix `error` in error variants, we can name this one as `ValueMismatch`